src/cli/index.ts

```diff
@@ -152,25 +160,34 @@ async function samplesFromDirectory(dataDir: string, topLevelRefs: string[] | un
         let jsonSamples: StringInput[] = [];
         const schemaSources: SchemaTypeSource[] = [];
         const graphQLSources: GraphQLTypeSource[] = [];
+        const typeScriptSources: TypeScriptTypeSource[] = [];
+
         for (const source of await readFilesOrURLsInDirectory(dir)) {
             // FIXME: We do a type switch here, but we know which types we're putting in
             // in the function above.  It should separate it right away.
             if (isJSONSource(source)) {
                 jsonSamples = jsonSamples.concat(source.samples);
             } else if (isSchemaSource(source)) {
                 schemaSources.push(source);
-            } else {
+            } else if (isGraphQLSource(source)) {
                 graphQLSources.push(source);
+            } else if (isTypeScriptSource(source)) {
+                typeScriptSources.push(source);
             }
```

Comment:
If you'd be so kind you could add a final `else` branch with `return panic("Unrecognized source");`

Reply:
Leave it for now, but in cases like this, where the compiler knows which values are possible, it's better to do

```
return assertNever(source.kind);
```

instead, which is a static check.